### PR TITLE
add algorithm Sieve of Eratosthenes

### PR DESCRIPTION
**Pull request by DONSIMON92**
* added Sieve of Eratosthenes
* add to the list in main file `README.md`

### CODE REVIEW

File: src/math/mod.rs [hunk 1]
```diff
@@ -15,11 +15,11 @@ pub use self::fast_power::fast_power;
 pub use self::greatest_common_divisor::{
     greatest_common_divisor_iterative, greatest_common_divisor_recursive,
 };
-pub use self::linear_sieve::LinearSieve;
```
Comment (er888kh) [inline src/math/mod.rs:18]: I think you removed this by accident.

Comment (DONSIMON92) [reply src/math/mod.rs]: thanks

Comment (er888kh) [reply src/math/mod.rs]: Looks good to me

**Review by er888kh — APPROVED** [review]

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**